### PR TITLE
DB-10926 Support timestamp(LONG VARCHAR) 

### PR DESCRIPTION
**Pull request by arnaud-splice**

### CODE REVIEW

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryComparisonOperatorNode.java [hunk 1]
```diff
@@ -55,154 +55,101 @@
...
+        TypeId rightTypeId = rightOperand.getTypeId();
+
+        if (! leftTypeId.isStringTypeId() && rightTypeId.isStringTypeId())
+        {
+            rightOperand = addCastNodeForStringToNonStringComparison(leftOperand, rightOperand);
+            rightTypeId = rightOperand.getTypeId();
+        }
+        else if (! rightTypeId.isStringTypeId() && leftTypeId.isStringTypeId())
+        {
+            leftOperand.addCastNodeForStringToNonStringComparison(rightOperand, leftOperand);
```
Comment (martinrupp) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryComparisonOperatorNode.java:150]: why is this line not the equivalent of 145? i would have expected sth like
`leftOperand = addCastNodeForStringToNonStringComparison(rightOperand, leftOperand);`

nice pulling out of a function!

Comment (hatyo) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryComparisonOperatorNode.java:150]: Also the result is ignored. Probably a bug?

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request closed without merging**